crates/nu-parser/src/parser.rs

```diff
@@ -1789,6 +1790,10 @@ pub fn parse_string_interpolation(working_set: &mut StateWorkingSet, span: Span)
 
 pub fn parse_variable_expr(working_set: &mut StateWorkingSet, span: Span) -> Expression {
     let contents = working_set.get_span_contents(span);
+    let input = match String::from_utf8(contents.to_vec()) {
+        Ok(str) => str,
+        Err(_) => "".to_owned(),
+    };
```

Comment:
This whole block is bad practice. In later usage you do not need the owned `String`, the `match` is nothing but `.unwrap_or_default()` for `str`/`String`. (Not sure if our Levenshtein matcher misbehaves when presented with the empty string.)

This should be either the `str::from_utf8` version or use the `String::from_utf8_lossy` function that returns a copy-on-write reference and only allocates for bad chars (the char replacement wouldn't matter that much in a fuzzy matching case)

Reply:
That's right 
Fixed

Reply:
Please try to read the full comment before marking it as resolved. I mentioned that the allocation as a `String` seems unnecessary.